### PR TITLE
✨ Mellow/Fly RRF E3 V1.0 (STM32F407VG)

### PR DESCRIPTION
**Pull request by belge96**
### Description

Since the Mellow/Fly RRF E3 V1.0 still wasn't supported i decided to add it myself, i thought, if i do add support i may aswell publish it for other people to have aswell (still had one of those boards around and didn't want to use RRF)

### Requirements

Mellow/Fly RRF E3 V1.0 (should work directly with a standard Ender 3 config for another E3* motherboard)

### Benefits

Added previously unsupported motherboard (incompatible pinout etc)

### Potential issues

This motherboard doesn't seem to have an EEPROM chip so it **NEEDS an SD card for EEPROM emulation**

### CODE REVIEW

**Comment by ellensp** [comment]
you have board_build.variant = MARLIN_FLY_RRF_E3_V1 

So there should be a folder buildroot/share/PlatformIO/variants/MARLIN_FLY_RRF_E3_V1 to define the new variant

**Comment by belge96** [comment]
> you have board_build.variant = MARLIN_FLY_RRF_E3_V1
> 
> So there should be a folder buildroot/share/PlatformIO/variants/MARLIN_FLY_RRF_E3_V1 to define the new variant

it uses marlin_STM32F407VGT6_CCM like BigTreeTech E3 RRF (which uses the same CPU/MCU) uses and doesn't have a separate definition either

the only difference between this and the BigTreeTech board is that it has an entirely different pinout and less ports (BTT has more optional ones)

**Comment by thisiskeithb** [comment]
> This motherboard doesn't seem to have an EEPROM chip so it NEEDS an SD card for EEPROM emulation

Flash-based emulation is preferred since it doesn’t require an SD card to be present.

File: Marlin/src/pins/stm32f4/pins_Fly_RRF_E3_V1.h [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+#if SD_CONNECTION_IS(ONBOARD)
+  #define ONBOARD_SDIO                            // Use SDIO for onboard SD
+  //#define SDIO_CLOCK                  48000000
+  #define SD_DETECT_PIN                     PA15
+#elif SD_CONNECTION_IS(CUSTOM_CABLE)
+  #error "SD CUSTOM_CABLE is not compatible with Fly RRF E3 V1."
+#endif
+
+// TODO ? board has inbuilt WiFi (ESP8266) but probably already runs a version of DuetWebServer if you've ever used it on RRF before and i have no idea what you'd need to flash it with to get it to work
```
Comment (thisiskeithb) [inline Marlin/src/pins/stm32f4/pins_Fly_RRF_E3_V1.h]: Review how this is done on other boards like the BTT E3 RRF: https://github.com/MarlinFirmware/Marlin/blob/f0b115a9993c0bf6eace56ad2b8c2d7944dfdd8a/Marlin/src/pins/stm32f4/pins_BTT_E3_RRF.h#L397-L406

**Comment by ellensp** [comment]
look at other examples

```
[Anet_ET4]
extends                     = stm32_variant
board                       = marlin_STM32F407VGT6_CCM
board_build.variant         = MARLIN_F4x7Vx    <- says to use files in buildroot/share/PlatformIO/variants/MARLIN_F4x7Vx
```

```
[env:BTT_E3_RRF]
extends             = stm32_variant
board               = marlin_STM32F407VGT6_CCM
board_build.variant = MARLIN_BTT_E3_RRF <- says to use files buildroot/share/PlatformIO/variants/MARLIN_BTT_E3_RRF
```

Yours

```
[env:FLY_RRF_E3_V1]
extends             = stm32_variant
board               = marlin_STM32F407VGT6_CCM
board_build.variant = MARLIN_FLY_RRF_E3_V1 <- says to use files buildroot/share/PlatformIO/variants/MARLIN_FLY_RRF_E3_V1
```

But you have not provided this variants directory so this line is invalid.

**Comment by ellensp** [comment]
the CI error is  due to the tight formatting constraints on board pins files.
you need to run "make format-pins -j"  or  buildroot/share/scripts/pinsformat.py to format your Marlin/src/pins/stm32f4/pins_Fly_RRF_E3_V1.h  file.

**Comment by thisiskeithb** [comment]
> the CI error is due to the tight formatting constraints on board pins files.

We also need this PR merged since it’s used for one of the tests:

- https://github.com/MarlinFirmware/Configurations/pull/1109

File: ini/stm32f4.ini [hunk 2]
```diff
@@ -194,6 +194,19 @@ build_flags         = ${stm32_variant.build_flags}
                       -DMF_RX_BUFFER_SIZE=255
                       -DMF_TX_BUFFER_SIZE=255
 
+#
+# Mellow/Fly RRF E3 V1 (STM32F407VGT6 ARM Cortex-M4)
+#
+[env:FLY_RRF_E3_V1]
+extends             = stm32_variant
+board               = marlin_STM32F407VGT6_CCM
+board_build.variant = MARLIN_FLY_RRF_E3_V1
```
Comment (thinkyhead) [inline ini/stm32f4.ini]: A custom variant is specified, but there's no `MARLIN_FLY_RRF_E3_V1` folder included. Do you have the variant files or know where to get them?

**Comment by thinkyhead** [comment]
This is about ready to merge, but we need to resolve the issue with the missing custom variant. If the `FLY_RRF_E3_V1` enviroment is truly identical to the `BTT_E3_RRF` then we can dump the `FLY_RRF_E3_V1` environment and just specify `BTT_E3_RRF` as the build environment for this board.

**Comment by thinkyhead** [comment]
This is still missing some pieces. Where is the best place to acquire them?

**Pull request merged**